src/legacy/ui/public/routes/route_manager.js

```diff
@@ -56,7 +56,22 @@ export default function RouteManager() {
     }
   };
 
-  self.run = function ($location, $route, $injector) {
+  self.run = function ($location, $route, $injector, $rootScope) {
+    if (window.elasticApm && typeof window.elasticApm.startTransaction === 'function') {
+      /**
+       * capture route-change events as transactions which happens after
+       * the browser's on load event.
+       *
+       * In Kibana app, this logic would run after the boostrap js files gets
+       * downloaded and get associated with the page-load transaction
+       */
+      $rootScope.$on('$routeChangeStart', (_, nextRoute) => {
```

Comment:
Any plans to also support this into the new platform?

Reply:
Hmmm good question, Do you know any Kibana app that is already running on New Platform? I can do a quick check.

Reply:
I think the things already running into the NP are here https://github.com/elastic/kibana/tree/master/x-pack/plugins and here https://github.com/elastic/kibana/tree/master/src/plugins

Reply:
I was able to see them getting tracked, so don't think any changes are required for the new platform.

Reply:
But there is a caveat to this, If the plugins inside new platform do Single page transitions, It would be hard to track them since they could use individual frameworks. So for now, we are only tracking apm plugin using our React Instrumentation. Would need something similar for other plugins as well. I will probably open a new PR to address those.

Reply:
in NP this will probably need to be hooked into `ApplicationService.navigateToApp`, However NP does not yet exposes that kind of hooks